src/components/ProgressForm/ProgressForm.tsx

```diff
@@ -29,13 +35,43 @@ const reducer = (state: progressStates, action: reducerAction) => {
 };
 
 const ProgressForm = ({ questions }: formProps) => {
+    const [isLoading, setIsLoading] = useState(false);
     const [state, dispatch] = useReducer(reducer, initialState);
     const manager = [state.progress, state.plan, state.blockers];
+    const [saveProgress] = useSaveProgressMutation();
+    const router = useRouter();
+
     const buttonSyle =
         state.progress && state.plan && state.blockers
             ? styles.buttonEnabled
             : styles.buttonDisabled;
 
+    if (isLoading) {
+        return <Loader></Loader>;
+    }
+
+    const handleSubmit = (e: MouseEvent) => {
```

Comment:
saveProgress function is asynchronous, so you need to use async and await to wait for the response before you can set the isLoading state to false

Reply:
since I am using `.then` thus this part can be skipped